services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -175,31 +183,21 @@ async def set_all_project_states(
         request.app, user_id, product_name, only_key_versions=True
     )
 
-    projects_list = []
-    project_types_list: List[bool] = []
-    if ptype in ("template", "all"):
-        template_projects = await db.load_template_projects(
-            user_id=user_id, filter_by_services=user_available_services
-        )
-
-        projects_list += template_projects
-        project_types_list += [True for i in range(len(template_projects))]
-
-    if ptype in ("user", "all"):  # standard only (notice that templates will only)
-        user_projects = await db.load_user_projects(
-            user_id=user_id, filter_by_services=user_available_services
-        )
-        projects_list += user_projects
-        project_types_list += [False for i in range(len(user_projects))]
-
-    start = int(request.query.get("start", 0))
-    count = int(request.query.get("count", len(projects_list)))
-
-    stop = min(start + count, len(projects_list))
-    projects_list = projects_list[start:stop]
-    project_types_list = project_types_list[start:stop]
-    await set_all_project_states(projects_list, project_types_list)
-    return {"data": projects_list}
+    projects, project_types, total_number_projects = await db.load_projects(
+        user_id=user_id,
+        filter_by_project_type=ProjectTypeAPI.to_project_type_db(project_type),
+        filter_by_services=user_available_services,
+        offset=offset,
+        limit=limit,
+    )
+    await set_all_project_states(projects, project_types)
+    return PageResponseLimitOffset.paginate_data(
+        data=projects,
+        request_url=request.url,
+        total=total_number_projects,
+        limit=limit,
+        offset=offset,
+    ).dict(by_alias=True, exclude_unset=True)
```

Comment:
Shouldn't we define this as a general policy for the web-server?? e.g. reusing  [RESPONSE_MODEL_POLICY](https://github.com/ITISFoundation/osparc-simcore/blob/18f60423183216149c75f310304ffff479aae1ee/services/web/server/src/simcore_service_webserver/catalog_api_handlers.py#L39)?

Reply:
ok moved RESPONSE_MODEL_POLICY to new rest_utils.py